artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/amq/AMQSession.java

```diff
@@ -95,6 +97,8 @@
 
    private final SimpleString clientId;
 
+   private final Set<Long> rollbackedIds = new ConcurrentHashSet<>();
```

Comment:
It could belong to `AMQConsumer` instead of here? That will make it less contended, because owned exclusively by the consumer

Reply:
This I can do. Thanks!